src/compiler/builder.ts

```diff
@@ -461,18 +550,39 @@ namespace ts {
          * in that order would be used to write the files
          */
         function emitNextAffectedFile(writeFile?: WriteFileCallback, cancellationToken?: CancellationToken, emitOnlyDtsFiles?: boolean, customTransformers?: CustomTransformers): AffectedFileResult<EmitResult> {
-            const affected = getNextAffectedFile(state, cancellationToken, computeHash);
+            let affected = getNextAffectedFile(state, cancellationToken, computeHash);
+            let isPendingEmitFile = false;
             if (!affected) {

```

Comment:
Could this whole block be moved into `getNextAffectedFile`? The logic seems a little out of place here. The main obstacle I see is that the return type would need to include `isPendingEmitFile`, or it would have to be calculated from `affected`.